chainer/dataset/dataset_mixin.py

```diff
@@ -0,0 +1,46 @@
+class DatasetMixin(object):
+
+    """Default implementation of dataset indexing.
+
+    DatasetMixin provides the :meth:`__getitem__` method using the per-example
+    addressing method :meth:`get_example` given by the implementation. This
+    mixin makes it easy to implement a new dataset that does not support
+    efficient slicing.
+
```

Comment:
We should note that the implementation that derives this mixin should implement `__len__` as well.